Subject: [PATCH] code review
---
Index: Client.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Client.h b/Client.h
--- a/Client.h	(revision def5bdd61d7cdae02612a4a2e6b03d21a16daa6c)
+++ b/Client.h	(date 1751466571773)
@@ -24,7 +24,7 @@
 private:
 	IO *io;
 	std::shared_ptr<Requests> requests;
-	Player *player = nullptr;
+    Player *player = nullptr; //лучше использовать unique_ptr
 
 	[[nodiscard]] auto get_ip() const -> IP;
 
Index: Client.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Client.cpp b/Client.cpp
--- a/Client.cpp	(revision def5bdd61d7cdae02612a4a2e6b03d21a16daa6c)
+++ b/Client.cpp	(date 1751466501318)
@@ -4,6 +4,8 @@
 #include "Log.h"
 #include "LoginData.h"
 
+#include <vector>
+
 Client::Client(IO *io):
 	io(io)
 {
@@ -54,6 +56,7 @@
 			this->login(packet);
 			break;
 		default:
+            //  лог не даёт информации о том, какой клиент вызвал событие
 			logger->warning("Unhandled packet type {}", packet.get_type());
 			break;
 	}
@@ -86,11 +89,11 @@
 	logger->info("Client {} bought item {}", this->player->id, item_id);
 }
 
-void Client::login(const client::Packet *packet)
+void Client::login(const client::Packet &packet) //несовпадение сигнатур, поправил метод
 {
-	uint64_t net_id = packet->L(0);
-	uint8_t net_type = packet->B(1);
-	const string &auth_key = packet->S(3);
+	uint64_t net_id = packet.L(0);
+	uint8_t net_type = packet.B(1);
+	const string &auth_key = packet.S(3);
 
 	if (net_type >= NetType::MAX_TYPE)
 	{
@@ -110,8 +113,9 @@
 	data.net_id = net_id;
 	data.net_type = net_type;
 
-	this->requests->add(&data, [&](const vector<Player*> &loaded) -> void
-	{
-		this->login_do(loaded[0], &data);
-	});
+	this->requests->add(
+			&data,
+			[this, data](const std::vector<Player *> &loaded) mutable {
+				this->login_do(loaded[0], &data);
+			});
 }
\ No newline at end of file
